src/react-console/index.js

```diff
@@ -0,0 +1,237 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import * as NovncLog from '@novnc/novnc/core/util/logging'
+import RFB from '@novnc/novnc/core/rfb'
+
+import classNames from 'classnames'
+import { Toolbar, noop } from 'patternfly-react'
+
+import VncActions from './VncActions'
+import constants from './constants'
+
+const { CONNECTED, CONNECTING, DISCONNECTED } = constants
+
+/* eslint no-console: ["warn", { allow: ["error"] }] */
+
+class VncConsole extends React.Component {
+  state = { status: CONNECTING }
+
+  addEventListeners () {
+    this.rfb.addEventListener('connect', this.onConnected)
+    this.rfb.addEventListener('disconnect', this.onDisconnected)
+    this.rfb.addEventListener('securityfailure', this.onSecurityFailure)
+  }
+
+  componentDidMount () {
+    const {
+      host,
+      port,
+      path,
+      encrypt,
+      resizeSession,
+      scaleViewport,
+      viewOnly,
+      shared,
+      credentials,
+      repeaterID,
+      vncLogging,
+      onInitFailed,
+    } = this.props
+
+    NovncLog.init_logging(vncLogging)
+    try {
+      const protocol = encrypt ? 'wss' : 'ws'
+      const url = `${protocol}://${host}:${port}/${path}`
+
+      const options = {
+        repeaterID,
+        shared,
+        credentials,
+      }
+
+      this.rfb = new RFB(this.novncElem, url, options)
+      this.addEventListeners()
+      this.rfb.viewOnly = viewOnly
+      this.rfb.scaleViewport = scaleViewport // if the remote session is smaller than HTML container, the view will be centered
+      this.rfb.resizeSession = resizeSession
+    } catch (e) {
+      onInitFailed && onInitFailed(e)
+      this.rfb = undefined
+    }
+  }
+
+  componentWillUnmount () {
+    this.disconnect()
+    this.removeEventListeners()
+    this.rfb = undefined
+  }
+
+  disconnect = () => {
```

Comment:
I know over in your patternfly PR (https://github.com/patternfly/patternfly-react/pull/1402#discussion_r258028551) @mareklibra asked why the function change from a class function to an arrow function.

AFAICT, in our project (web-ui), we don't have the same babel level or transforms that auto bind the class functions, but their project does.  So, it is necessary to keep it that way here, but not there.  From a spot check, we're on the same `babel` version, but they are on the `env@^1.6.1` preset and we're on the `latest@6.24.1` preset.

Net result - probably do not need to keep that change on the patternfly-react PR.

Reply:
Yes, I know and problem not only in `this`, but also if look at original code, there are some methods defined like arrow, some - not. And on my opinion this is not consistent.

Reply:
It was just a thought.  Either way, it'll get resolved in patternfly-react and our local copy of the component can get removed.